server/src/fetchers/relationship-fetchers.js

```diff
@@ -0,0 +1,100 @@
+// @flow
+
+import type { Viewer } from '../session/viewer';
+import {
+  type RelationshipErrors,
+  undirectedStatus,
+  directedStatus,
+} from 'lib/types/relationship-types';
+
+import _groupBy from 'lodash/fp/groupBy';
+import invariant from 'invariant';
+
+import { dbQuery, SQL } from '../database';
+
+type RelationshipOperation =
+  | 'delete_directed'
+  | 'friend'
+  | 'pending_friend'
+  | 'know_of';
+type UserRelationshipOperations = {
+  [string]: $ReadOnlyArray<RelationshipOperation>,
+};
+
+async function fetchFriendRequestRelationshipOperations(
+  viewer: Viewer,
+  userIDs: string[],
+) {
+  const query = SQL`
+    SELECT user1, user2, status
+    FROM relationships_directed
+    WHERE
+      (user1 IN (${userIDs}) AND user2 = ${viewer.userID}) OR
+      (user1 = ${viewer.userID} AND user2 IN (${userIDs}))
+    UNION SELECT user1, user2, status
+    FROM relationships_undirected
+    WHERE
+      (user1 = ${viewer.userID} AND user2 IN (${userIDs})) OR
+      (user1 IN (${userIDs}) AND user2 = ${viewer.userID})
+  `;
+
+  const [result] = await dbQuery(query);
+
+  const relationshipsByUserId = _groupBy(
+    ({ user1, user2 }) => (user1.toString() === viewer.userID ? user2 : user1),
+    result,
+  );
+
+  const errors: RelationshipErrors = {};
+  const userRelationshipOperations: UserRelationshipOperations = {};
+  for (const userID in relationshipsByUserId) {
+    const relationships = relationshipsByUserId[userID];
+
+    if (relationships.length === 3) {
```

Comment:
I implemented it like this since `relationships` can include one of the following relationship rows:

[know_of, blocked_1, blocked_2]
[know_of, blocked_1]
[know_of, blocked_2]
[know_of, pending_friend_1]
[know_of, pending_friend_2]
[know_of, friends]
[know_of]